src/EmotiBitWiFiHost.cpp

```diff
@@ -72,27 +70,166 @@ int8_t EmotiBitWiFiHost::begin()
 	isStartingConnection = false;
 
 	dataThread = new std::thread(&EmotiBitWiFiHost::updateDataThread, this);
-
+	ofResetElapsedTimeCounter(); //resets timer so nothing is off sync
+	ipSearchTimer = ofGetElapsedTimeMillis();
+	advertizingTimer = ipSearchTimer;
 	return SUCCESS;
 }
 
+void EmotiBitWiFiHost::getAvailableSubnets() {
+	vector<string> ips;
+	const int NUM_TRIES_GET_IP = 10;
+	int tries = 0;
+	while (ips.size() <= 0 && tries < NUM_TRIES_GET_IP)
+	{
+		ips = getLocalIPs();
+		ofSleepMillis(100);
+		tries++;
+	}
+	if (ips.size() > 0) {
+		//get all available subnets
+		for (int subnet = 0; subnet < ips.size(); subnet++)
+		{
+			vector<string> ipSplit = ofSplitString(ips.at(subnet), ".");
+			string tempSubnet = ipSplit.at(0) + "." + ipSplit.at(1) + "." + ipSplit.at(2);
+				if (ofFind(availableSubnets, tempSubnet) == availableSubnets.size()) {
+					availableSubnets.push_back(tempSubnet);
+				}
+		}
+	}
+}
+
+void EmotiBitWiFiHost::getAdvertisingIpSend() {
+	if (enableBroadcast) { //disabled for now, implement as a future feature
+		//initially tries broadcasting
+		for (int ip = 0; ip < availableSubnets.size(); ip++) {
+			string braodcastIp = availableSubnets.at(ip) + "." + ofToString(255);
+			ipSearchCxn.Connect(braodcastIp.c_str(), advertisingPort);
+			ipSearchCxn.SetEnableBroadcast(true);
+			ipSearchCxn.SetNonBlocking(true);
+			ipSearchCxn.SetReceiveBufferSize(pow(2, 10));
+			// Send advertising message
+			string packet = EmotiBitPacket::createPacket(EmotiBitPacket::TypeTag::HELLO_EMOTIBIT, advertisingPacketCounter++, "", 0);
+			//ofLogVerbose() << "Sent: " << packet;
+			ipSearchCxn.Send(packet.c_str(), packet.length());
+		}
+	}
+	if(advertisingIps.size() == 0 || !enableBroadcast) { //connects through unicasting
+		if (emotibitSubnets.size() == 0) { //initial search through all subnets
+			getAvailableSubnets(); //if subnet appears after osciliscope is open (mobile hotspot)
+			for (int ip = 0; ip < availableSubnets.size(); ip++) {
+				for (int hostId = 0; hostId < 255; hostId++) {
+					string braodcastIp = availableSubnets.at(ip) + "." + ofToString(hostId);
+					ipSearchCxn.Connect(braodcastIp.c_str(), advertisingPort);
+					ipSearchCxn.SetEnableBroadcast(true);
+					ipSearchCxn.SetNonBlocking(true);
+					ipSearchCxn.SetReceiveBufferSize(pow(2, 10));
+					// Send advertising message
+					string packet = EmotiBitPacket::createPacket(EmotiBitPacket::TypeTag::HELLO_EMOTIBIT, advertisingPacketCounter++, "", 0);
+					//ofLogVerbose() << "Sent: " << packet;
+					ipSearchCxn.Send(packet.c_str(), packet.length());
+				}
+			}
+		}
+		else {
+			for (int hostId = 0; hostId < 255; hostId++) {
+				string braodcastIp = emotibitSubnets.at(0) + "." + ofToString(hostId);
```

Comment:
This should be fine, given that the assumption of all EmotiBits connect to the same network, holds true. I guess its OK for now, but we should probably make an issue check for all available subnets for emotibit addition to the network.

+

typo on broadcastIp